DSharpPlus/Entities/Interaction/Application/DiscordApplicationCommand.WeakEquals.cs

```diff
@@ -0,0 +1,282 @@
+                    && (option.AutoComplete ?? false) == (other.AutoComplete ?? false)
+                    && BoxedIntegersMatch(option.MinValue, other.MinValue)
+                    && BoxedIntegersMatch(option.MaxValue, other.MaxValue)
+                    && EnumListsMatch(option.ChannelTypes, other.ChannelTypes)
+                    && ChoicesMatch(option.Choices, other.Choices)
+                    && LocalizationsMatch(option.NameLocalizations, other.NameLocalizations)
+                    && LocalizationsMatch(option.DescriptionLocalizations, other.DescriptionLocalizations)
+                    && OptionsMatch(option.Options, other.Options);
+
+                if (!match)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static bool ChoicesMatch
+    (
+        IReadOnlyList<DiscordApplicationCommandOptionChoice>? a,
+        IReadOnlyList<DiscordApplicationCommandOptionChoice>? b
+    )
+    {
+        if ((a is null || a.Count == 0) && (b is null || b.Count == 0))
+        {
+            return true;
+        }
+
+        if (a is null || b is null)
+        {
+            return false;
+        }
+
+        if (a.Count != b.Count)
+        {
+            return false;
+        }
+
+        foreach (DiscordApplicationCommandOptionChoice choice in a)
+        {
+            if (!b.Any(c =>
+                {
+                    return c.Name == choice.Name 
+                        && ChoiceValuesMatch(c, choice)
+                        && LocalizationsMatch(c.NameLocalizations, choice.NameLocalizations);
+                })
+            )
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static bool ChoiceValuesMatch(object a, object b)
+    {
+        return a switch
+        {
+            int value => b is int other && value == other,
+            long value => b is long other && value == other,
+            double value => b is double other && value == other,
+            string value => b is string other && value == other,
+            _ => false
+        };
+    }
+
+    private static bool IntegrationTypesMatch
+    (
+        IReadOnlyList<DiscordApplicationIntegrationType>? a,
+        IReadOnlyList<DiscordApplicationIntegrationType>? b
+    )
+    {
+        if 
+        (
+            (a is null || a.Count == 0  || a is [DiscordApplicationIntegrationType.GuildInstall]) 
+            && (b is null || b.Count == 0 || b is [DiscordApplicationIntegrationType.GuildInstall])
+        )
+        {
+            return true;
+        }
+
+        if (a is null || b is null)
+        {
+            return false;
+        }
+
+        if (a.Count != b.Count)
+        {
+            return false;
+        }
+
+        foreach (DiscordApplicationIntegrationType type in a)
```

Comment:
Consider SequenceEqual (they should be the same, but at worse, Intersect().Count > 0?)

Both would eliminate the above if check too

Reply:
SequenceEqual is inapplicable because the types need not be ordered, this ends up being a faster but more verbose version of checking via LINQ